File: backend/Shared/EventPlanR.Application/Dto/Event/EventDetailsDto.cs
using EventPlanr.Application.Dto.Common;
using EventPlanr.Domain.Enums;

namespace EventPlanr.Application.Dto.Event;

public class EventDetailsDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public EventCategory Category { get; set; }
    public DateTimeOffset FromDate { get; set; }
    public DateTimeOffset ToDate { get; set; }
    public string Venue { get; set; } = null!;
    public AddressDto Address { get; set; } = null!;
    public CoordinatesDto Coordinates { get; set; } = null!;
    public string? Description { get; set; }
    public string? CoverImageUrl { get; set; }
}

File: frontend/lib/data/network/models/event_planr/event/event_details_dto.dart
import 'package:event_planr_app/data/network/models/event_planr/common/address_dto.dart';
import 'package:event_planr_app/data/network/models/event_planr/common/coordinates_dto.dart';
import 'package:event_planr_app/data/network/models/event_planr/news_post/news_post_dto.dart';
import 'package:event_planr_app/data/network/models/event_planr/organization/organization_dto.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'event_details_dto.freezed.dart';

part 'event_details_dto.g.dart';

@freezed
class EventDetailsDto with _$EventDetailsDto {
  const factory EventDetailsDto({
    required String id,
    required String name,
    required String? description,
    required String? coverImageUrl,
    required int category,
    required DateTime fromDate,
    required DateTime toDate,
    required String venue,
    required AddressDto address,
    required CoordinatesDto coordinates,
    required OrganizationDto organization,
    required NewsPostDto latestNews,
  }) = _EventDetailsDto;

  factory EventDetailsDto.fromJson(Map<String, Object?> json) =>
      _$EventDetailsDtoFromJson(json);
}
